include/add_pfunit_sources.cmake

```diff
@@ -24,7 +24,7 @@ function( ADD_PFUNIT_SOURCES out_var )
 
          # replace the extension with .F90 to determine the output file name
 	 get_filename_component (extension ${file} EXT)
-         string( REGEX REPLACE "${extension}" ".F90" out_file "${CMAKE_CURRENT_BINARY_DIR}/${rel_file}" )
+         string( REPLACE "${extension}" ".F90" out_file "${CMAKE_CURRENT_BINARY_DIR}/${rel_file}" )
```

Comment:
The problem was that the extension includes a period char, which is a wildcard in regex patterns. So if you try building in a subdirectory names, say, `pfunit`, then `/pf` will match `.pf` and then cmake will look for a directory like `/tmp.F90unit....` and try to create it.

Reply:
Seems like this change is still not sufficient for other cases.    

At a minimum, I think the line above the change should be LAST_EXT instead of EXT.  But even then we should try to guard against substitutions elsewhere in the name.   E.g.

"/tmp/my.pf/foo.pf" should change to "/tmp/my.pf/foo.pf"

Maybe something like:

```cmake
get_filename_component (extension ${file} LAST_EXT)
string( REGEX REPLACE "\${extension}$" ".F90" out_file"${CMAKE_CURRENT_BINARY_DIR}/${rel_file}" )
```

My cmake rexexp-foo is weak, so I'd need to try variations to see how to best escape the "." for the suffix and keep the "$" check for the end.

Does this make sense?

Reply:
You're absolutely right that we should be using `LAST_EXT`. Let me look a bit more. I still think that we probably want `string( REPLACE` not `string(REGEX REPLACE` but I'll search around and make this more robust according to your suggestions.

Reply:
(and maybe this is the source of my other woes)

Reply:
Since there is no mechanism to `string( REPLACE` only the last occurance, I have pushed a new commit that takes into consideration your feedback. I'm still not thrilled by the idea of using a string harvested from the OS as a regex, but I couldn't see a straightforward way to ensure that the replacement happens only if it matches the last few characters of file with the full path... actually, I just thought of a more robust solution... another commit should be coming in a moment.